cvat/apps/engine/serializers.py

```diff
@@ -405,17 +405,76 @@ def update(self, instance, validated_data):
             instance.bug_tracker)
         instance.subset = validated_data.get('subset', instance.subset)
         labels = validated_data.get('label_set', [])
-        for label in labels:
-            LabelSerializer.update_instance(label, instance)
+        if instance.project_id is None:
+            for label in labels:
+                LabelSerializer.update_instance(label, instance)
+        validated_project_id = validated_data.get('project_id', None)
+        if validated_project_id is not None and validated_project_id != instance.project_id:
+            project = models.Project.objects.get(id=validated_data.get('project_id', None))
+            if instance.project_id is None:
+                for old_label in instance.label_set.all():
+                    try:
+                        new_label = project.label_set.filter(name=old_label.name).first()
+                    except ValueError:
+                        raise serializers.ValidationError(f'Target project does not have label with name "{old_label.name}"')
+                    old_label.attributespec_set.all().delete()
```

Comment:
In some specific cases we want to transfer attributes as well. For example, when we have identical definition for labels and attributes. Are you going to support that?

Reply:
Yes, but it's quite an independent feature so in future implementations